dynamics/client.py

```diff
@@ -82,25 +82,36 @@ class DynamicsClient:  # pylint: disable=R0904,R0902
         status.HTTP_503_SERVICE_UNAVAILABLE: WebAPIUnavailable,
     }
 
-    def __init__(self, api_url: str, token_url: str, client_id: str, client_secret: str, scope: List[str]):
+    def __init__(self, api_url: str, token_url: str, client_id: str, client_secret: str,
+                 scope: Optional[List[str]], resource: Optional[str]):
```

Comment:
I would add None as the default argument for both `scope` and `resource`, so that only one of them needs to be passed when initializing.

Reply:
Done.